sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/text/TextSuite.scala

```diff
@@ -65,6 +65,14 @@ class TextSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("support for partitioned reading") {
+    val df = spark.read.format("text").load(testDir)
+    val data = df.filter("year = '2015'").select("value").collect()
+
+    assert(data(0) == Row("2015-test"))
+    assert(data.length == 1)
```

Comment:
I'm new to this codebase (even new to Scala). But I'm happy to give that a go in a separate PR :)